pkg/kube/translation/annotations/redirect.go

```diff
@@ -37,8 +41,15 @@ func (r *redirect) PluginName() string {
 func (r *redirect) Handle(e Extractor) (interface{}, error) {
 	var plugin apisixv1.RedirectConfig
 	plugin.HttpToHttps = e.GetBoolAnnotation(_httpToHttps)
+	plugin.URI = e.GetStringAnnotation(_uri)
+
+	retCode, err := strconv.Atoi(e.GetStringAnnotation(_retCode))
 	// To avoid empty redirect plugin config, adding the check about the redirect.
-	if plugin.HttpToHttps {
+	if err == nil {
+		plugin.RetCode = retCode
+		return &plugin, nil
```

Comment:
Why return it directly?

Reply:
Cannot be used alone `ret_code`? the `ret_ code 'needs to be used with' uri ' ？ If so, that's not right.

Reply:
yes, if there is no redirect target, then how can there be a redirect status code?